packages/core/src/State.ts

```diff
@@ -19,6 +19,12 @@ import { StateNode } from './StateNode';
 import { getMeta, nextEvents } from './stateUtils';
 import { initEvent } from './actions';
 
+type Equals<A1 extends any, A2 extends any> = (<A>() => A extends A2
+  ? 1
+  : 0) extends <A>() => A extends A1 ? 1 : 0
+  ? 1
+  : 0;
```

Comment:
😳

Reply:
Could this be `true` or `false` instead of `1` or `0`?

Reply:
It could. It's just taken from [here](https://github.com/millsp/ts-toolbelt/blob/319e55123b9571d49f34eca3e5926e41ca73e0f3/sources/Any/Equals.ts) and this type collection is using the 1/0 pattern in all its types.

I'm still wondering if we should use this though, see the comment here: https://github.com/statelyai/xstate/pull/2436#discussion_r672664520